src/Screens/ConsultantProfile.tsx

```diff
@@ -0,0 +1,438 @@
+  const [rateError, setRateError] = useState('');
+  const [experienceError, setExperienceError] = useState('');
+  const [descriptionError, setDescriptionError] = useState('');
+  const [categoryError, setCategoryError] = useState('');
+  const [cityError, setCityError] = useState('');
+  const [imageError, setImageError] = useState('');
+  const [loading, setLoading] = useState(false);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (user) {
+      setRate(user?.rate || 0);
+      setAge(user?.age || 0);
+      setGender(user?.gender || 'male');
+      setDescription(user?.description || '');
+      setCategory(user?.category || '');
+      setExperience(user?.experience || 0);
+      setCity(user?.city || '');
+      setProfilePicture(user?.profilePicture || '');
+    }
+  }, [user]);
+
+  const handleImageUpload = async () => {
+    const result: any = await launchImageLibrary({mediaType: 'photo'});
+    if (!result?.didCancel) {
+      let name = JSON.stringify(new Date().getTime());
+      const reference = storage().ref(`images/${name}`);
+      const task = reference.putFile(result?.assets[0]?.uri);
+      task.on('state_changed', taskSnapshot => {
+        console.log(
+          `${taskSnapshot.bytesTransferred} transferred out of ${taskSnapshot.totalBytes}`,
+        );
+      });
+      task.then(async () => {
+        await storage()
+          .ref(`images/${name}`)
+          .getDownloadURL()
+          .then(async (url: any) => {
+            setProfilePicture(url);
+          })
+          .catch(e => console.error(e));
+      });
+    }
+  };
+
+  const categoryOptions = categoriesList.map(category => ({
+    label: category.name,
+    value: category.name,
+  }));
+
+  const handleUpdate = () => {
+    const ageError = isValidNumber(age, 'age');
+    const rateError = isValidNumber(rate, 'rate');
+    const experienceError = isValidNumber(experience, 'experience');
+    const descriptionError = isValidDescription(description);
+    let categoryError = '';
+    let cityError = '';
+    let imageError = '';
+    if (!category) {
+      categoryError = 'Category is required';
+    }
+    if (!city) {
+      cityError = 'City is required';
+    }
+    if (!profilePicture) {
+      imageError = 'Profile Pic is required';
+    }
+    if (
+      ageError ||
+      rateError ||
+      experienceError ||
+      categoryError ||
+      cityError ||
+      descriptionError
+    ) {
+      setAgeError(ageError);
+      setRateError(rateError);
+      setExperienceError(experienceError);
+      setDescriptionError(descriptionError);
+      setCategoryError(categoryError);
+      setCityError(cityError);
+      setImageError(imageError);
+      return;
+    } else {
+      setLoading(true);
+      setAgeError('');
+      setRateError('');
+      setExperienceError('');
+      setDescriptionError('');
+      setCategoryError('');
+      setCityError('');
+      setImageError('');
+      const updatedObject = {

```

Comment:
make this object in state instead of using many separate states. As i describe you.